storage/storage.go

```diff
@@ -51,18 +51,30 @@ type Storage interface {
 	// filtered.
 	ListBuckets(ctx context.Context, prefix string) ([]Bucket, error)
 
-	// MakeBucket creates given bucket.
-	MakeBucket(ctx context.Context, bucket string) error
+	// Make creates bucket for remote operations and dir/file for local.
+	Make(ctx context.Context, opts MakeOpts) (ReadCloserFile, error)
+
+	// Open opens the given source. Return value can be either a readable and/or writable.
+	Open(ctx context.Context, src *url.URL) (ReadCloserFile, error)
```

Comment:
While `fs.Open()` actually opens a file, `s3.Open()` retrieves S3 object and reads its content. I suspect sharing a common interface for these functions might be confusing. Instead, maybe we can keep separate two methods for fs and s3. What do you think?

Reply:
Yeah, I also think `fs.Open()` and `s3.Open()` do different things, and there is similar problem with 
`fs.Make()` and `s3.Make()`. First I thought of introducing new methods as you have suggested, but in that case we would have an interface with 11 methods, where `s3` does not implement(returns `notimplemented` error) 2 methods while `fs` not implements 4 methods. This reinforces the issue of making one interface implementation not be totally replaceable with the other.

One option might be to change storage interface as to contain only the methods common for both `s3` / `fs` and make clients work directly with `s3` or `fs` objects. This removes client side interface-abstraction, but it also removes surprises. Currently we are making an implicit assumption that for `storage.Storage{ local filesystem }`, users of interface will not call `storage.ListBuckets`. And in the future if `s5cmd` supports more s3 operations, dissimilarity of `s3` and `fs` will increase more.

So, shall we remove/modify `Open`, `Make` (also `ReadCloserFile`) and introduce new functions?

Reply:
let's discuss this later :) We can keep `Open()` and `Make()` for now.